src/nix/formatter.cc

```diff
@@ -72,6 +94,12 @@ struct CmdFormatterRun : MixFormatter, MixJSON
         auto evalState = getEvalState();
         auto evalStore = getEvalStore();
 
+        auto flakeRef = parseFlakeRef(fetchSettings, ".", std::filesystem::current_path().string());
```

Comment:
The flakeref should be obtained from the installable returned by `parseInstallable()` below. `flakeRef.input` has a method `getSourcePath()` that returns the physical path of a flake, if any.

Reply:
Thanks, @xokdvium! I've updated the code to use the helper method you pointed me to. I'm not sure about this nullptr check you suggested, though? I think it's impossible to [construct a `ref` with a null pointer](https://github.com/NixOS/nix/blob/0db10fc875d3dafe365d157aed7a6d9344092fc2/src/libutil/include/nix/util/ref.hh#L10-L11).

Reply:
Great! Marking this thread resolved.